pontoon/api/schema.py

```diff
@@ -111,16 +112,16 @@ class Meta:
     )
 
     def resolve_localizations(obj, _info, include_disabled, include_system):
-        qs = obj.project_locale
-
-        records = qs.filter(project__disabled=False, project__system_project=False)
+        records = (
+            obj.project_locale.visible_for(_info.context.user)
+            .filter(
+                Q(project__disabled=include_disabled)
+                | Q(project__system_project=include_system)
+            )
+            .distinct()
+        )
```

Comment:
I fell into a similar trap:
https://github.com/mozilla/pontoon/pull/1639#discussion_r431067188

If you'll check results, you'll notice it doesn't really work. We could also add tests to catch this.

Reply:
Okay, I see my flaw in understanding this problem. I think that without a testing matrix, anybody can make the same mistake.
I'll create a test and then try to figure if that's solvable in an ORM query.

Reply:
@mathjazz I added two tests to reduce the risk of that regression.